sycl/include/sycl/ext/oneapi/work_group_static.hpp

```diff
@@ -0,0 +1,134 @@
+//==----- work_group_static.hpp --- SYCL group local memory extension -----==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include <sycl/access/access.hpp>             // for address_space, decorated
+#include <sycl/detail/defines_elementary.hpp> // for __SYCL_ALWAYS_INLINE
+#include <sycl/exception.hpp>                 // for exception
+#include <sycl/ext/intel/usm_pointers.hpp>    // for multi_ptr
+#include <sycl/ext/oneapi/properties/properties.hpp> // for properties
+
+#include <type_traits> // for enable_if_t
+
+namespace sycl {
+inline namespace _V1 {
+namespace ext::oneapi {
+namespace experimental {
+
+#ifdef __SYCL_DEVICE_ONLY__
+#define __SYCL_WG_SCOPE [[__sycl_detail__::wg_scope]]
+#else
+#define __SYCL_WG_SCOPE
+#endif
+
+/// @brief Allocate data in device local memory.
+/// Any work_group_static object will be place in device local memory and hold
+/// an object of type T. work_group_static object are implicitly treated as
+/// static.
+/// @tparam T must be a trivially constructible and destructible type
+template <typename T> class __SYCL_WG_SCOPE work_group_static {
+public:
+  __SYCL_ALWAYS_INLINE work_group_static() = default;
+  work_group_static(const work_group_static &) = delete;
+  work_group_static &operator=(const work_group_static &) = delete;
+
+  operator T &() noexcept { return *getDecorated(); }
+
+  template <class TArg = T>
+  typename std::enable_if<!std::is_array_v<TArg>, work_group_static &>::type
```

Comment:
```suggestion
  typename std::enable_if_t<!std::is_array_v<TArg>, work_group_static &>
```

but if you've already made it a template (instead of partial specialization, possibly via a mixin), why not `<typename TArg = T, typename = std::enable_if_t<>>`?